rotkehlchen/balances/historical.py

```diff
@@ -1,29 +1,187 @@
+import logging
 from collections import defaultdict
-from typing import TYPE_CHECKING, TypedDict
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, TypedDict
 
+from rotkehlchen.accounting.constants import DEFAULT, EVENT_CATEGORY_MAPPINGS, EXCHANGE
 from rotkehlchen.assets.asset import Asset
-from rotkehlchen.constants import ZERO
+from rotkehlchen.constants import DAY_IN_SECONDS, ONE, ZERO
 from rotkehlchen.constants.prices import ZERO_PRICE
-from rotkehlchen.db.filtering import HistoryEventFilterQuery
-from rotkehlchen.db.history_events import DBHistoryEvents
+from rotkehlchen.db.filtering import (
+    DBFilter,
+    DBMultiStringFilter,
+    DBNestedFilter,
+    DBNotEqualFilter,
+    DBTimestampFilter,
+)
+from rotkehlchen.errors.asset import UnknownAsset
 from rotkehlchen.errors.misc import NotFoundError, RemoteError
 from rotkehlchen.errors.price import NoPriceForGivenTimestamp
+from rotkehlchen.errors.serialization import DeserializationError
 from rotkehlchen.fval import FVal
-from rotkehlchen.history.events.structures.types import EventDirection
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.history.events.structures.types import (
+    EventDirection,
+    HistoryEventSubType,
+    HistoryEventType,
+)
 from rotkehlchen.history.price import PriceHistorian
-from rotkehlchen.types import Timestamp
-from rotkehlchen.utils.misc import ts_ms_to_sec
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_fval, deserialize_timestamp
+from rotkehlchen.types import Timestamp, TradeType
+from rotkehlchen.utils.misc import timestamp_to_daystart_timestamp
 
 if TYPE_CHECKING:
     from rotkehlchen.db.dbhandler import DBHandler
-    from rotkehlchen.history.events.structures.base import HistoryBaseEntry
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
 
 
 class HistoricalBalance(TypedDict):
     amount: FVal
     price: FVal
 
 
+@dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=True)
```

Comment:
why introduce and maintain another data structure for trades and for history events? Just to save memory?

It does not make sense to me.

Also if you ever wanted to do that that would mean you would need to introduce this at the source. And have Trades inherit this. And history events inherit the other one, no ? So you at least have a common data struct for both? And then inherit from minimal and inheirt from normal?

Otherwise you are duplicating structs here.

Please explain the logic. Do not just go about making changes in response to this comment before discussing this.

But in general this whole approach seems hacky.